src/shared/components/submit-button.tsx

```diff
@@ -1,17 +1,18 @@
 import React from "react";
 import LockIcon from "../icons/lock.svg";
+import { useAuthoredState, useInteractiveState } from "@concord-consortium/lara-interactive-api";
 import css from "../styles/helpers.scss";
 
 interface IProps {
-  authoredState: { required?: boolean } | undefined;
-  interactiveState: { submitted?: boolean } | undefined;
-  setInteractiveState: ((interactiveState: {submitted?: boolean}) => void) | undefined;
   isAnswered: boolean;
 }
 
 // This hook can be used by any interactive that defines `required` property in its authored state and
 // `submitted` property in its interactive state (student state).
-export const SubmitButton: React.FC<IProps> = ({ authoredState, interactiveState, setInteractiveState, isAnswered }) => {
+export const SubmitButton: React.FC<IProps> = ({ isAnswered }) => {
```

Comment:
I'm not sure it is worth it, but it'd be cool if the SubmitButton was passed the authored state type and the interactiveState type.  Then it could cause a type error if these types don't have `required?` and `submitted?`. 

It looks like this is possible based on this post: https://mariusschulz.com/blog/passing-generics-to-jsx-elements-in-typescript

Reply:
My first implementation was like this. SubmitButton wasn't extracted back then, but basically useRequiredQuestion would complain if there were no `required?/submitted?` props. But then I changed it to make it less strict, since some interactives/question might not want to support the submitt button (or hint or other features that we add later).

Now, they can still use the same `BaseQuestionApp` without any configuration. `requiored?` ensures that required prop might be defined, but it doesn't have to. If it's not, it's all good and the feature is disabled.

This typing actually more helps to write `useRequiredQuestion` or `SubmitButton`, as we're adding some (loose) assumptions about passed in states.

I can change it to be more strict again, but then we would need to add more configuration to `BaseQuestionApp` and add `disabled` props to most of the hooks. Or don't add `disabled` and assume that `BaseQuestionApp` is forcing developer to support these features (since most questions actually should support submit button, hint, etc.).